backend/src/main/java/fk/prof/backend/aggregator/ProfileWorkInfo.java

```diff
@@ -0,0 +1,81 @@
+package fk.prof.backend.aggregator;
+
+import com.koloboke.collect.map.hash.HashObjIntMap;
+import com.koloboke.collect.map.hash.HashObjIntMaps;
+import recording.Recorder;
+
+import java.time.LocalDateTime;
+import java.util.*;
+
+/**
+ * Non getter methods in this class are not thread-safe
+ * But, this should not be a concern because instances of this class are maintained for every work id
+ * Updates to state associated with a work id happens in context of a request (single thread)
+ */
+public class ProfileWorkInfo {
+  private AggregationStatus status = AggregationStatus.SCHEDULED;
+  private LocalDateTime startedAt = null, endedAt = null;
+  private int samples = 0;
```

Comment:
Samples should probably be a type-scoped thing. Because samples can have different value for CPU and off-CPU sampled recording. So it should be scoped by type. No?